my-app/src/index.js

```diff
@@ -38,6 +45,21 @@ class Game extends Component {
       stepNumber: history.length,
       xIsNext: !this.state.xIsNext
     });
+
+    console.log('componentDidUpdate', squares);
+    const board = history;
+
+    this.aiInstance.play(board).then(pos => {
+
+      console.log('AI plays on the position '+pos);
+
+      this.handleClick(pos);
```

Comment:
@W1zardDude why didn't you put this to getDerivedStateFromProps?
seems you make a recursion call without stop